### PR TITLE
Limit Range Config Abstraction

### PR DESCRIPTION
**Pull request by whwalter**
All handlers will have to source a config from the limit ranges in the namespace for a resource so this wraps that behavior in an interface and provides an implementation.

More concretely this type will receive a LimitRangeLister from an informer created during controller creation time and can be passed to the handlers as a shared resource. 

I didn't provide a constructor since there is only a single member that is an interface type that can be assigned.

### CODE REVIEW

**Review by dskatz — COMMENTED** [review]
Should this instead use an Informer to consistently sync LimitRanges for all namespaces?  

If I am reading this implementation correctly, if there is a change to a LimitRange that would not be reflected `NewConfig`.

An Informer could constantly sync LimitRangers of type Container to a local cache and then queried with the Lister per mutation.

**Comment by whwalter** [comment]
The idea would be for an informer to provide the lister implementation via the Lister() interface on the informer but this abstracts the lister implementation away from the type completely. This type only cares that it has a LimitRangeLister implementation vs having to have its own informer that then generates a Lister of its own. 

The informer can be constructed a single time during controller creation and the same informer can provide a single LimitRangeLister to a single LimitRanger implementation that is then passed to handlers.

File: internal/admission/limitrange.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package admission
+
+import (
+	"github.com/kanopy-platform/hedgetrimmer/internal/limitrange"
+	corev1Listers "k8s.io/client-go/listers/core/v1"
+)
+
+type LimitRanger interface {
+	corev1Listers.LimitRangeLister
+	NewConfig(namespace string) (*limitrange.Config, error)
```
Comment (dskatz) [inline internal/admission/limitrange.go]: A constructor style naming seems odd in an interface.

```suggestion
	LimitRangeConfig(namespace string) (*limitrange.Config, error)
```

File: internal/limitrange/limitrange.go [hunk 2]
```diff
@@ -17,3 +21,29 @@ func NewConfig(lri corev1.LimitRangeItem, resource corev1.ResourceName) Config {
...
+
+//LimitRange provides an implementation of the LimitRanger interface defined in admission. This implementation is designed to provider a generic config sourcing tool for all mutation handlers.
+type LimitRange struct {
+	lister corev1Listers.LimitRangeLister
+}
+
+//LimitRangeConfig takes a namespace string and returns a Config for memory or a nil if no limit range of type Container is found in the namespace. It returns a non-nil error if there is an error sourcing data from the cluster api or the namespace name is empty
+func (lr *LimitRange) LimitRangeConfig(namespace string) (*Config, error) {
+	if namespace == "" {
+		return nil, fmt.Errorf("Invalid namespace: %q", namespace)
```
Comment (dskatz) [inline internal/limitrange/limitrange.go:33]: nit: should be lower case
```suggestion
		return nil, fmt.Errorf(invalid namespace: %q", namespace)
```

**Review by dskatz — APPROVED** [review]
lgtm

**Pull request merged**